test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,307 @@
+				Expect(err).NotTo(HaveOccurred())
+
+				err = shootTestOperations.K8sShootClient.DeleteStatefulSet(helmDeployNamespace, "redis-master")
+			}
+			cleanupGuestbook()
+			cleanupRedis()
+
+			shootTestOperations.ShootGardenerTest.Logger.Info("redis and the guestbook app have been cleaned up!")
+		}()
+
+		helm := Helm(ResourcesDir)
+		err := shootTestOperations.EnsureDirectories(helm)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = shootTestOperations.DownloadAndDeployHelmChart(ctx, helm, helmDeployNamespace, "stable/redis")
+		Expect(err).NotTo(HaveOccurred())
+
+		err = shootTestOperations.WaitUntilPodIsRunning(ctx, "redis-master-0", helmDeployNamespace, shootTestOperations.K8sShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		redisSlaveLabels := map[string]string{
+			"app":  "redis",
+			"role": "slave",
+		}
+
+		err = shootTestOperations.WaitUntilPodWithLabelsIsReady(ctx, redisSlaveLabels, helmDeployNamespace, shootTestOperations.K8sShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		guestBookParams := struct {
+			HelmDeployNamespace string
+			ShootDNSHost        string
+		}{
+			helmDeployNamespace,
+			fmt.Sprintf("guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain),
+		}
+
+		// Parse the guestbook app template
+		var writer bytes.Buffer
+		err = guestBooktpl.Execute(&writer, guestBookParams)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Apply the guestbook app resources to shoot
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = shootTestOperations.K8sShootClient.Applier().ApplyManifest(context.TODO(), manifestReader)
+		Expect(err).NotTo(HaveOccurred())
+
+		// define guestbook app urls
+		guestBookAppURL := fmt.Sprintf("http://guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain)
+		pushUrl := fmt.Sprintf("%s/rpush/guestbook/foobar-%s", guestBookAppURL, shootTestOperations.Shoot.Info.Name)
+		pullUrl := fmt.Sprintf("%s/lrange/guestbook", guestBookAppURL)
+
+		// Check availability fo the guestbookapp
+		err = shootTestOperations.WaitUntilGuestbookAppIsAvailable(ctx, []string{guestBookAppURL, pushUrl, pullUrl})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Push foobar-<shoot-name> to the guestbook app
+		_, err = shootTestOperations.HTTPGet(ctx, pushUrl)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Pull foobar
+		pullResponse, err := shootTestOperations.HTTPGet(ctx, pullUrl)
+		Expect(err).NotTo(HaveOccurred())
+
+		responseBytes, err := ioutil.ReadAll(pullResponse.Body)
+		Expect(err).NotTo(HaveOccurred())
+
+		// test if foobar-<shoot-name> was pulled successfully
+		bodyString := string(responseBytes)
+		Expect(bodyString).To(ContainSubstring(fmt.Sprintf("foobar-%s", shootTestOperations.Shoot.Info.Name)))
+
+		shootTestOperations.ShootGardenerTest.Logger.Info("Guestbook app was deployed successfully!")
+
+	}, GuestbookAppTimeout)
+
+	CIt("Dashboard should be available", func(ctx context.Context) {
+		err := shootTestOperations.DashboardAvailable(ctx)
```

Comment:
Tests should be isolated from each other, i.e. order of execution should _not_ influence the outcome of a test. This test is not separated. Please make sure to check *all* your tests for isolation to avoid yet another round of reviews.

Reply:
As discussed, this comment does not apply here